rust-runtime/aws-smithy-client/src/retry.rs

```diff
@@ -296,13 +308,17 @@ where
     Handler: Clone,
     R: ClassifyResponse<SdkSuccess<T>, SdkError<E>>,
 {
-    type Future = Pin<Box<dyn Future<Output = Self> + Send + Sync>>;
+    type Future = Pin<Box<dyn Future<Output = Self> + Send>>;
 
     fn retry(
         &self,
         req: &Operation<Handler, R>,
         result: Result<&SdkSuccess<T>, &SdkError<E>>,
     ) -> Option<Self::Future> {
+        let sleep = match &self.sleep {
+            Some(sleep) => sleep,
+            None => return None,
```

Comment:
Should we log a debug trace here to make it easier to diagnose why retry isn't working if something is misconfigured?